diesel/src/migration/mod.rs

```diff
@@ -49,7 +63,27 @@ impl<'a> Migration for &'a Migration {
     fn revert(&self, conn: &SimpleConnection) -> Result<(), RunMigrationsError> {
         (&**self).revert(conn)
     }
+
     fn file_path(&self) -> Option<&Path> {
         (&**self).file_path()
     }
+
+    fn metadata(&self) -> Option<&Metadata> {
+        (&**self).metadata()
+    }
+}
+
+/// Represents metadata associated with a migration.
+///
+/// The format of a migration's metadata is dependent on the migration format
+/// being used.
+///
+/// For Diesel's built in SQL file migrations, metadata is stored in a file
+/// called `metadata.toml`. Diesel looks for a single key, `run_in_transaction`.
+/// By default, all migrations are run in a transaction on SQLite and
+/// PostgreSQL. This behavior can be disabled for a single migration by setting
+/// this to `false`.
+pub trait Metadata {
+    /// Get the metadata at the given key, if present
+    fn get(&self, key: &str) -> Option<Cow<str>>;
```

Comment:
I think I'd like a more strongly typed API here. Something like:

```rust
fn get<T>(&self, key: &str) -> Option<Result<T, MigrationError>>
where
    T: serde::DeserializeOwned;
```

The TOML impl can then be:

```rust
fn get<T>(&self, key: &str) -> Option<Result<T, MigrationError>>
where
    T: DeserializeOwned,
{
    self.0.get(key)
        .map(|v| v.try_into::<T>().map_err(|e| MigrationError::InvalidMetadata(Box::new(e))))
}
```

Then we can drop the manual boolean parsing code, and more complex consumers can tightly couple themselves to deeper data structures if they want.